File: UM/CSC419_SPRING2026/hw_2/hw2_bounds.rb
# Create array of 10 integers (initialized to nil)
arr = Array.new(10, 0)

begin
  # Accessing an element out of range
  new_elem = arr[10]
  puts "Array element at index 10: #{new_elem}"
rescue IndexError => e
  puts "An error occurred: #{e.class}: #{e.message}"
end

File: UM/CSC419_SPRING2026/hw_2/hw2_bounds.py
def main():
    try:
        # Create array of size 10
        arr = [0] * 10
        # Access out of bounds element
        out_of_bounds = arr[10]
    except Exception as e:
        # This is how we grab the excepts "name"
        print(f"An error occurred, Error Type : {e.__class__.__name__} - {e}")


if __name__ == "__main__":
    main()
